lib/array_intersection.rb

```diff
@@ -1,4 +1,11 @@
 def intersection(list1, list2)
```

Comment:
This works, but is O(n^2) for time complexity.  Can you see a way to do this using a Hash?